src/actions/company.js

```diff
@@ -515,3 +525,166 @@ export const queryCompanyWorkExperiences = ({
     dispatch(setWorkExperiences(companyName, getError(error)));
   }
 };
+
+const subscribeCompany = ({ companyName }) => async (dispatch, getState) => {
+  const state = getState();
+  const token = tokenSelector(state);
+  const box = companyIsSubscribedBoxSelectorByName(companyName)(state);
+  if (!isFetched(box) || !box.data) {
+    return;
+  }
+  const { companyId } = box.data;
+  if (!companyId) {
+    return;
+  }
+
+  dispatch(
+    setIsSubscribed(
+      companyName,
+      getFetched({
+        isSubscribed: true,
+        companyId,
+      }),
+    ),
+  );
+  try {
+    const data = await subscribeCompanyApi({
+      companyId,
+      token,
+    });
+    const {
+      subscribeCompany: { success },
+    } = data;
+
+    if (!success) {
+      dispatch(
+        setIsSubscribed(
+          companyName,
+          getFetched({
+            isSubscribed: false,
+            companyId,
+          }),
+        ),
+      );
+    }
+  } catch (error) {
+    dispatch(
+      setIsSubscribed(
+        companyName,
+        getFetched({
+          isSubscribed: false,
+          companyId,
+        }),
+      ),
+    );
+    throw error;
+  }
+};
+
+const unsubscribeCompany = ({ companyName }) => async (dispatch, getState) => {
+  const state = getState();
+  const token = tokenSelector(state);
+  const box = companyIsSubscribedBoxSelectorByName(companyName)(state);
+  if (!isFetched(box) || !box.data) {
+    return;
+  }
+  const { companyId } = box.data;
+  if (!companyId) {
+    return;
+  }
+
+  dispatch(
+    setIsSubscribed(
+      companyName,
+      getFetched({
+        isSubscribed: false,
+        companyId,
+      }),
+    ),
+  );
+  try {
+    const data = await unsubscribeCompanyApi({
+      companyId,
+      token,
+    });
+    const {
+      unsubscribeCompany: { success },
+    } = data;
```

Comment:
Optional/不一定要改：感覺可以再把 success 包進去
```suggestion
    const success = await unsubscribeCompanyApi({
      companyId,
      token,
    });
```
```js
export const unsubscribeCompanyApi = ({ companyId, token }) =>
  const data = await graphqlClient({
    query: unsubscribeCompanyGql,
    token,
    variables: { input: { companyId } },
  });
  return data.unsubscribeCompany.success
}
```

Reply:
我覺得這樣改很棒，也不用再解構，已改，謝謝建議